fault_proof/src/lib.rs

```diff
@@ -262,4 +303,160 @@ where
+            let parent_game_address = self
+                .fetch_game_address_by_index(U256::from(parent_game_index))
+                .await?;
+            let parent_game = OPSuccinctFaultDisputeGame::new(
+                parent_game_address,
+                l1_provider_with_wallet.clone(),
+            );
+
+            Ok((
+                parent_game.status().call().await?.status_ != GameStatus::IN_PROGRESS,
+                oldest_game_address,
+            ))
+        }
+    }
+
+    /// Attempts to resolve a challenged game.
+    ///
+    /// This function checks if the game is in progress and challenged, and if so, attempts to resolve it.
+    async fn try_resolve_games(
+        &self,
+        index: U256,
+        mode: Mode,
+        l1_provider_with_wallet: L1ProviderWithWallet<F, P, T>,
+        l2_provider: L2Provider,
+    ) -> Result<()> {
+        let game_address = self.fetch_game_address_by_index(index).await?;
+        let game = OPSuccinctFaultDisputeGame::new(game_address, l1_provider_with_wallet.clone());
+        if game.status().call().await?.status_ != GameStatus::IN_PROGRESS {
+            tracing::info!(
+                "Game {:?} at index {:?} is not in progress, not attempting resolution",
+                game_address,
+                index
+            );
+            return Ok(());
+        }
+
+        let claim_data = game.claimData().call().await?.claimData_;
+        match mode {
+            Mode::Proposer => {
+                if claim_data.status != ProposalStatus::Unchallenged {
+                    tracing::info!(
+                        "Game {:?} at index {:?} is not unchallenged, not attempting resolution",
+                        game_address,
+                        index
+                    );
+                    return Ok(());
+                }
+            }
+            Mode::Challenger => {
+                if claim_data.status != ProposalStatus::Challenged {
+                    tracing::info!(
+                        "Game {:?} at index {:?} is not challenged, not attempting resolution",
+                        game_address,
+                        index
+                    );
+                    return Ok(());
+                }
+            }
+        }
+
+        let current_timestamp = l2_provider
+            .get_l2_block_by_number(BlockNumberOrTag::Latest)
+            .await?
+            .header
+            .timestamp;
+        let deadline = U256::from(claim_data.deadline).to::<u64>();
+        if deadline >= current_timestamp {
+            tracing::info!(
+                "Game {:?} at index {:?} deadline {:?} has not passed, not attempting resolution",
+                game_address,
+                index,
+                deadline
+            );
+            return Ok(());
+        }
+
+        let contract = OPSuccinctFaultDisputeGame::new(game_address, l1_provider_with_wallet);
+        let receipt = contract.resolve().send().await?.get_receipt().await?;
```

Comment:
Add a note here that we should wait for some required confirmations.

Reply:
I just added required confirmations and timeout for sends